### PR TITLE
Add barometric sensor

### LINKED ISSUE #30: HumiditySensor + BarometricSensor
I just got a xiaomi "weather" sensor which has temperature, humidity, and barometric pressure.

It would be nice to be include TemperatureSensor, HumiditySensor and BarometricSensor in the list of @types so that the user can choose the most important to them.

### PR DESCRIPTION
**Pull request by tim-hellhake**
Resolves #30

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -202,8 +204,27 @@ <h3>Events</h3>
           </tr>
         </table>
       </section>
-      
-            
+      <section id="BarometricSensor">
```
[the remark this answers is not in the document]
Comment (tim-hellhake) [reply index.html]: I used the name from #30.
But `BarometricPressureSensor` makes indeed more sense.

**Comment by mrstegeman** [comment]
Please open a gateway issue to support this.

**Comment by mrstegeman** [comment]
Nevermind, done: https://github.com/mozilla-iot/gateway/issues/2577

File: index.html [hunk 2]
```diff
@@ -796,6 +817,19 @@ <h2>AlarmProperty</h2>
...
+      <section id="BarometricPressureProperty">
+        <h2>BarometricPressureProperty</h2>
+        <table>
+          <tr>
+            <th>type</th>
+            <td>number/integer</td>
+          </tr>
+          <tr>
+            <th>unit</th>
+            <td>hectopascal</td>
```
Comment (benfrancis) [inline index.html:830]: I don't think hectopascal is an SI [base](https://en.wikipedia.org/wiki/SI_base_unit) or [derived](https://en.wikipedia.org/wiki/SI_derived_unit) unit like all the other units in schemas are. Should this be [pascal](https://en.wikipedia.org/wiki/Pascal_(unit)) instead?

**Comment by madb1lly** [comment]
Hi @benfrancis,

Hectopascal is equivalent to millibars so is a standard unit for barometric pressure, even though not strictly an SI unit.

I think it makes sense for schemas to permit standard units of all kinds, not only SI ones.

Cheers 🙂

**Comment by benfrancis** [comment]
@madb1lly Unfortunately that's not what the specification says so conforming clients would not understand the unit. See https://github.com/mozilla-iot/schemas/pull/65#pullrequestreview-473394749

**Comment by madb1lly** [comment]
Hello @benfrancis,

Nonetheless, hecto is a permitted SI prefix denoting 10^2, so hPa is an acceptable SI unit even if Pa is the base unit. I don't see where in the Thing description spec SI prefixes are not permitted? In fact one example gives units of milliseconds, which is the same type of usage of an SI unit.

On that basis it seems that hPa is totally acceptable, isn't it?

**Comment by benfrancis** [comment]
@madb1lly Having looked at the [original source](https://www.bipm.org/utils/common/pdf/si-brochure/SI-Brochure-9.pdf) (also educating myself!) I think you're right. "hecto" is listed in Table 7 (SI prefixes). Also, good point about milliseconds.

This does mean that writing a general purpose WoT client might be a bit trickier than I thought as far as units are concerned, as the number of possible units explodes! But it is at least consistent with the specification. Sorry for the false alarm :)

**Comment by tim-hellhake** [comment]
I considered using Pa instead of hPa but because of the `milliseconds` example, I thought prefixes would be ok 😬

**Comment by madb1lly** [comment]
> This does mean that writing a general purpose WoT client might be a bit trickier than I thought as far as units are concerned, as the number of possible units explodes! But it is at least consistent with the specification. Sorry for the false alarm :)

Yes, unit conversion difficulties are (almost) as old as a hills: https://www.mentalfloss.com/article/25845/quick-6-six-unit-conversion-disasters

However, if SI units are used then it shouldn't be too difficult. I did a quick web search and found several existing JSON structures for representing SI units. It shouldn't be too difficult for someone with the skills (i.e. not me!) to use one for the gateway, or any other WebThing explorer.

**Pull request merged**